### PR TITLE
Useful HandlerWithKey middleware

### PR DESCRIPTION
**Pull request by halabooda**
Possibility to use HandlerWithKey function and bit refactoring

### CODE REVIEW

File: stampede.go [hunk 1]
```diff
@@ -123,11 +123,3 @@ func BytesToHash(b ...[]byte) uint64 {
 	}
 	return d.Sum64()
 }
-
-func StringToHash(s ...string) uint64 {
```
Comment (pkieltyka) [inline stampede.go]: Why remove this method? I can’t accept the PR that removes this method. Other part of PR looks good tho, I’ll check it in detail when I’m at a computer

Comment (halabooda) [reply stampede.go]: Oh, I removed the old default function usage; and removed the function as used only there

Comment (halabooda) [reply stampede.go]: returned to use

File: _example/with-key.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		time.Sleep(3 * time.Second)
+
+		w.WriteHeader(200)
+		w.Write([]byte(r.Header.Get("Authorization")))
+	})
+
+	http.ListenAndServe(":3333", r)
+}
+
+func StringToHash(s ...string) uint64 {
```
Comment (pkieltyka) [inline _example/with-key.go:89]: please remove from example and use the function from the package

Comment (halabooda) [reply _example/with-key.go:89]: done

**Pull request merged**